packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/04-upgradeable-plugin/04-subsequent-builds.md

```diff
@@ -0,0 +1,141 @@
+    account = _account;
+  }
+}
+```
+
+Builds that you publish don't necessarily need to introduce new storage varaibles of your contracts and don't necessarily need to change the storage. To read more about Upgradeability, check out [OpenZeppelin's UUPSUpgradeability implementation here](https://docs.openzeppelin.com/contracts/4.x/api/proxy#UUPSUpgradeable).
+
+### 3. Alternatively, a build implementation modifying bytecode
+
+Updates for your contracts don't necessarily need to affect the storage, they can also modify the plugin's bytecode. Modifying the contract's bytecode, means making changes to:
+
+- functions
+- constants
+- immutables
+- events
+- errors
+
+For this third build, then, we want to change the bytecode of our implementation as an example, so we 've introduced two separate permissions for the `storeNumber` and `storeAccount` functions and named them `STORE_NUMBER_PERMISSION_ID` and `STORE_ACCOUNT_PERMISSION_ID` permission, respectively. Additionally, we decided to add the `NumberStored` and `AccountStored` events as well as an error preventing users from setting the same value twice. All these changes only affect the contract bytecode and not the storage.
+
+Here, it is important to remember how Solidity stores `constant`s (and `immutable`s). In contrast to normal variables, they are directly written into the bytecode on contract creation so that we don't need to worry that the second `bytes32` constant that we added shifts down the storage so that the value in `uint256 public number` gets lost.
+It is also important to note that, the `initializeFromBuild2` could be left empty. Here, we just emit the events with the currently stored values.
+
+```solidity
+import {PluginUUPSUpgradeable, IDAO} '@aragon/osx/core/plugin/PluginUUPSUpgradeable.sol';
+
+/// @title SimpleStorage build 3
+contract SimpleStorageBuild3 is PluginUUPSUpgradeable {
+  bytes32 public constant STORE_NUMBER_PERMISSION_ID = keccak256('STORE_NUMBER_PERMISSION'); // changed in build 3
+  bytes32 public constant STORE_ACCOUNT_PERMISSION_ID = keccak256('STORE_ACCOUNT_PERMISSION'); // added in build 3
+
+  uint256 public number; // added in build 1
+  address public account; // added in build 2
+
+  // added in build 3
+  event NumberStored(uint256 number);
+  event AccountStored(address number);
+  error AlreadyStored();
+
+  /// @notice Initializes the plugin when build 3 is installed.
+  function initializeBuild3(
+    IDAO _dao,
+    uint256 _number,
+    address _account
+  ) external reinitializer(3) {
+    __PluginUUPSUpgradeable_init(_dao);
+    number = _number;
+    account = _account;
+
+    emit NumberStored({number: _number});
+    emit AccountStored({account: _account});
+  }
+
+  /// @notice Initializes the plugin when the update from build 2 to build 3 is applied.
+  /// @dev The initialization of `SimpleStorageBuild2` has already happened.
+  function initializeFromBuild2() external reinitializer(3) {
```

Comment:
The contracts have been updated recently and the tutorials should be adapted as well with the new source files
https://github.com/aragon/simple-storage-example-plugin/tree/main/contracts

Reply:
Does this follow-up task exist?

Reply:
I would personally follow this model here, already:

https://github.com/aragon/simple-storage-example-plugin/blob/main/contracts/release1/build3/SimpleStorageR1B3.sol#L30-L50

Reply:
Yes - task exist within Jira and all comments have  been added to this Notion doc for tracking: https://www.notion.so/aragonorg/Updating-Simple-Storage-example-in-OSx-docs-9716e7e9723e4caea0dc183fb301dade?pvs=4

Reply:
Yes - this has been added to comments for the Jira task!